src/stravalib/tests/unit/test_model.py

```diff
@@ -260,7 +272,7 @@ def test_entity_collections(self):
                 },
             ]
         }
-        a.from_dict(d)
+        a = model.Athlete.model_validate(d)
```

Comment:
removes legacy behavior and uses pydantic 2.x which prefers immutability and returns a new object with the data ingested.